pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/TopicConsumerConfigurationData.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl.conf;
+
+import java.io.Serializable;
+import java.util.regex.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class TopicConsumerConfigurationData implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    @NonNull
+    private Pattern topicsPattern;
+    private int priorityLevel;
+
+    public boolean matchesTopicName(String topicName) {
+        if (topicsPattern == null || topicName == null) {
+            return false;
+        }
+        return topicsPattern.matcher(topicName).matches();
+    }
+
+    public static TopicConsumerConfigurationData of(@NonNull String topicNameOrPattern,
+                                                    ConsumerConfigurationData<?> conf) {
+        return of(topicNameOrPattern, conf.getPriorityLevel());
```

Comment:
If I understand correctly, do we pass the `ConsumerConfigurationData` because we want to extend this feature in the future?

Reply:
Yes, this is the 'default' constructor - the topic configuration will initialise with a default configuration matching that of the `ConsumerConfigurationData`.

Reply:
Got it, thanks!